index.js

```diff
@@ -1,30 +1,131 @@
 const express = require('express');
 
-// const jsonfile = require('jsonfile');
-
-/**
- * ===================================
- * Configurations and set up
- * ===================================
- */
+const jsonfile = require('jsonfile');
+const file = 'pokedex.json';
 
 // Init express app
 const app = express();
 
-/**
- * ===================================
- * Routes
- * ===================================
- */
-
-app.get('*', (request, response) => {
-  // send response with some data (a string)
-  response.send(request.path);
-});
-
-/**
- * ===================================
- * Listen to requests on port 3000
- * ===================================
- */
+
+var getType = (request, response) => {
+	
+	// app.get('/type/:types', getType)
+	// :type is a parameter
+	// localhost:3000/type/grass   ;   grass is basically the argument
+	// request.params.types --------> output: grass
+
+	jsonfile.readFile(file, (err, obj) => {
```

Comment:
The order of the call back matters, You code will work.
But it will do a jsonfile read every time there is a server request.

Check out some of your class mate's code. They only read it once at the top and then use the data multiple times.